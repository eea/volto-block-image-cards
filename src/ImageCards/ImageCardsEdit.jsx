import React from 'react';
import { SidebarPortal } from '@plone/volto/components'; // EditBlock
import InlineForm from '@plone/volto/components/manage/Form/InlineForm';
import config from '@plone/volto/registry';

import ImageCardsView from './ImageCardsView';

import ImageCardSchema from './schema';

const tweakSchema = (schema, data) => {
  const { blockRenderers } = config.blocks.blocksConfig.imagecards;
  const extension = data.display
    ? blockRenderers[data.display].schemaExtender
    : null;
  return extension ? extension(schema, data) : schema;
};

const ImageCardsEdit = (props) => {
  const basicSchema = ImageCardSchema(props);
  const schema = tweakSchema(basicSchema, props.data);
  const display = props.data.display || 'carousel';
  const CardsView =
    config.blocks.blocksConfig.imagecards.blockRenderers?.[display]?.edit ||
    ImageCardsView;

  return (
    <>
      <CardsView data={props.data} />

      <SidebarPortal selected={props.selected}>
        <InlineForm
          schema={schema}
          title={schema.title}
          onChangeField={(id, value) => {
            props.onChangeBlock(props.block, {
              ...props.data,
              [id]: value,
            });
          }}
          formData={{
            ...props.data,
            display: props.data?.display || 'carousel',
          }}
          block={props.block}
        />
      </SidebarPortal>
    </>
  );
};

export default ImageCardsEdit;
