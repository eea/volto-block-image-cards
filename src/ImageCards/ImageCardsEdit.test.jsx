import React from 'react';
import renderer from 'react-test-renderer';
import config from '@plone/volto/registry';
import ImageCardsEdit from './ImageCardsEdit';

config.blocks.blocksConfig = {
  imagecards: {
    blockRenderers: {
      id1: {
        schemaExtender: (schema, data) => ({
          ...schema,
          properties: {
            ...schema.properties,
            customProperty: { type: 'string', title: 'Custom Property' },
          },
        }),
      },
      id2: {
        view: () => <div>id2</div>,
      },
      id3: {
        view: () => <div>id3</div>,
      },
    },
  },
};

describe('ImageCardsEdit', () => {
  it('should render without crashing', () => {
    const component = renderer.create(
      <ImageCardsEdit
        data={{ display: 'id1' }}
        selected={false}
        onChangeBlock={jest.fn()}
        block={{}}
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render without crashing and no display', () => {
    const component = renderer.create(
      <ImageCardsEdit
        data={{ display: undefined }}
        selected={false}
        onChangeBlock={jest.fn()}
        block={{}}
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
