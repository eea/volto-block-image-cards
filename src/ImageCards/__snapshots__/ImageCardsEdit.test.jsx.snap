// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCardsEdit should render without crashing 1`] = `
<div
  className="image-cards-wrapper"
>
  
</div>
`;

exports[`ImageCardsEdit should render without crashing and no display 1`] = `
<div
  className="image-cards-wrapper"
>
  
</div>
`;
